fluss-flink/fluss-flink-common/src/test/java/com/alibaba/fluss/flink/source/FlussSourcePKITCase.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (c) 2025 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.fluss.flink.source;
+
+import com.alibaba.fluss.client.table.Table;
+import com.alibaba.fluss.config.ConfigOptions;
+import com.alibaba.fluss.flink.row.RowConverters;
+import com.alibaba.fluss.flink.source.deserializer.Order;
+import com.alibaba.fluss.flink.source.deserializer.OrderPartial;
+import com.alibaba.fluss.flink.source.deserializer.OrderPartialDeserializationSchema;
+import com.alibaba.fluss.flink.source.deserializer.RowDataDeserializationSchema;
+import com.alibaba.fluss.flink.source.enumerator.initializer.OffsetsInitializer;
+import com.alibaba.fluss.flink.source.testutils.FlinkTestBase;
+import com.alibaba.fluss.flink.source.testutils.MockDataUtils;
+import com.alibaba.fluss.metadata.Schema;
+import com.alibaba.fluss.metadata.TableDescriptor;
+import com.alibaba.fluss.metadata.TablePath;
+import com.alibaba.fluss.row.GenericRow;
+import com.alibaba.fluss.row.InternalRow;
+import com.alibaba.fluss.types.RowType;
+
+import org.apache.flink.api.common.eventtime.WatermarkStrategy;
+import org.apache.flink.streaming.api.datastream.DataStreamSource;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.StringData;
+import org.apache.flink.types.RowKind;
+import org.apache.flink.util.CloseableIterator;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+import static com.alibaba.fluss.testutils.DataTestUtils.row;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+/** Integration tests for the {@link FlussSource} class with PK tables. */
+public class FlussSourcePKITCase extends FlinkTestBase {
```

Comment:
We don't need a separate ITCase class for testing PK table. Please merge it with `FlussSourceLogITCase` into `FlussSourceITCase`. This can help us reduce testing time. 

The `BeforeEach` method can be extracted as a private method and be called in the beginning of testing method.